services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,19 +44,54 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  } else if(thermalMgrQueueHandle == NULL) {
+    return ERR_CODE_INVALID_STATE;
+  }
+  
+  if(xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 0) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event = { .type = THERMAL_MGR_EVENT_OS_OVER_TEMP };
+  xQueueSend(thermalMgrQueueHandle, &event, (TickType_t) 10);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t event;
+  error_code_t errCode;
+  float temp;
```

Comment:
Previously requested changes: Move the scope of temp down to right before its use

Reply:
Do we do this to conserve memory in case the event.type is not any of the ones we care about?

Reply:
Yeah basically